cli/src/org/partiql/cli/functions/QueryDDB.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
+package org.partiql.cli.functions
+
+import com.amazon.ion.system.IonReaderBuilder
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
+import org.partiql.lang.eval.EvaluationSession
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueFactory
+import org.partiql.lang.eval.stringValue
+import org.partiql.lang.types.FunctionSignature
+import org.partiql.lang.types.StaticType
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder
+import com.amazonaws.services.dynamodbv2.document.ItemUtils
+import com.amazonaws.services.dynamodbv2.model.ExecuteStatementRequest
+
+/**
+ * [QueryDDB] is a function to perform a single query on DDB using PartiQL.
+ * It uses the default configuration provider to establish a connection with AWS. Please reference the official
+ * AWS documentation for specifying which account/profile to use via credentials overrides. Reference the CLI.md
+ * file within this repository for more information.
+ */
+internal class QueryDDB(valueFactory: ExprValueFactory) : BaseFunction(valueFactory) {
+
+    private val client: AmazonDynamoDB = AmazonDynamoDBClientBuilder.defaultClient()
+
+    override val signature = FunctionSignature(
+        name = "query_ddb",
```

Comment:
At some point, we will want to namespace things. Also, right now the functions must be manually added to the compiler pipeline. We should consider java/jvm service loader rather than modifying CLI directly. This would enable customers to provide their own functions

Reply:
That being said, this is a good start. I want to overhaul function resolution after my bag op work

Reply:
Agreed. Service loader will be the best path forward.